src/gui/shell.cpp

```diff
@@ -725,9 +725,23 @@ void Shell::handleNeovimNotification(const QByteArray &name, const QVariantList&
 			m_mouseHide = variant_not_zero(args.at(1));
 			int val = m_mouseHide ? 1 : 0;
 			m_nvim->api0()->vim_set_var("GuiMousehide", val);
-		} else if (guiEvName == "Close" && args.size() == 1) {
+		} else if (guiEvName == "Close" && (args.size() == 1 || args.size() == 2)) {
 			qDebug() << "Neovim requested a GUI close";
-			emit neovimGuiCloseRequest();
+
+			// Decide which exit status Neovim-qt should exit with.
+			int status;
```

Comment:
Provide a default value, even something like `{}` is fine.

This helps prevent undefined behavior bugs.